compiler/rustc_const_eval/src/util/alignment.rs

```diff
@@ -23,33 +23,79 @@ where
 
     let ty = place.ty(local_decls, tcx).ty;
     let unsized_tail = || tcx.struct_tail_for_codegen(ty, typing_env);
-    match tcx.layout_of(typing_env.as_query_input(ty)) {
-        Ok(layout)
+
+    // Try to normalize the type to resolve any generic parameters
+    let normalized_ty = match tcx.try_normalize_erasing_regions(typing_env, ty) {
+        Ok(normalized) => normalized,
+        Err(_) => {
+            // If normalization fails, fall back to the original type
+            ty
```

Comment:
why? it feels a lot safer to conservatively return `true` here, same as when `layout_of` fails